doc/lang.md

```diff
@@ -1754,6 +1755,39 @@ to be disabled.
 
 *Mode:* Run.
 
+#### Expect
+
+The operator `expect` has the following syntax:
+
+```scala
+A.expect(P)
+expect(A, P)
+```
+
+The left-hand side `A` must be an action or a run. The right-hand side `P` must
+be a non-action Boolean expression.
+
+The semantics of this operator is as follows:
+
+- Evaluate action `A`:
+  - When `A`'s result is `false`, emit a runtime error.
+  - When `A`'s result is `true`:
+    - Commit the variable updates.
+    - Evaluate `P`:
+      - If `P` evaluates to `false`, emit a runtime error (similar to `assert`).
+      - If `P` evaluates to `true`, rollback to the previous state and return `true`.
```

Comment:
Does 'rollback to the previous state' mean that the 'committed' variable updates are undone? So `A.then(B).expect(P)` would end up in the state after `A` has been applied but before `B`?

Reply:
Yeah, it's exactly that. Not really rolling back to a state. I will fix it here too.

Reply:
I have fixed the description